python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+
+    It identifies the most important subgraph from the original graph that plays a
+    critical role in GNN-based graph classification.
+
+    It employs Monte Carlo tree search (MCTS) in efficiently exploring different subgraphs
+    for explanation and uses Shapley values as the measure of subgraph importance.
+
+    Parameters
+    ----------
+    model: nn.Module
+        The GNN model to explain.
+
+        * The required arguments of its forward function are `DGLGraph` graph and its
+        node features.
+        * The output of its forward function is the logits for the predicted
+          graph classes.
+    num_gnn_layers: int
+        Number of layers of GNN model. Needed to calculate the l-hop neighbouring
+        nodes of graph. Computing shapley values for big and complex graphs
+        is time and resource consuming. The shapley algorithm implemented in the paper
+        efficiently approximate shapley values by finding coalition of l-hop neighboring
+        nodes of the graph, where the GNN model has l layers.
+    coef: float
+        The hyperparameter that controls the trade-off between exploration and exploitation.
+        A higher exploration rate encourages exploration of relatively unvisited nodes.
+        So, the more a node has been visited (and the more certain is its total reward value),
+        the less it will be visited. For example, if a node has been visited many times and
+        has a high total reward value, a high coef will cause the algorithm to explore other nodes
+        instead of continuing to visit that node.
+    high2low: bool, optional
+        Pruning action either "High2low" or "Low2high" (refer to paper).
+        "High2low": Whether to expand children nodes from high degree to low degree when
+        extend the child nodes in the search tree. "Low2high" is opposite of "High2low".
+        If True, it will use "High2low" pruning action, default to True.
+    num_child_expand: int, optional
+        Max number of children a tree node is allowed to have/expand to.
+    max_iter: int, optional
+        Max number of iterations for MCTS.
+    node_min: int, optional
+        The leaf threshold node number.
+    mc_sampling_steps: int
+        Monte carlo sampling steps.
+    """
+
+    def __init__(
+        self,
+        model,
+        num_gnn_layers,
+        coef=10.0,
+        high2low=True,
+        num_child_expand=2,
+        max_iter=20,
+        node_min=3,
+        mc_sampling_steps=100,
+    ):
+        super(SubgraphXExplainer, self).__init__()
+        self.num_gnn_layers = num_gnn_layers
+        self.coef = coef
+        self.high2low = high2low
+        self.num_child_expand = num_child_expand
+        self.max_iter = max_iter
+        self.node_min = node_min
+        self.mc_sampling_steps = mc_sampling_steps
+
+        self.shapley = shapley
+
+        self.model = model
+        self.model.eval()
+
+    def get_value_func(self, feat, **kwargs):
```

Comment:
No need to have an additional function, just move the implementation to `explain_graph`. It is used just once anyway.

Reply:
done.